src/main/java/com/google/sps/data/SessionContext.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.User;
+import com.google.sps.data.NonPersistentUserRepository;
+import com.google.sps.data.UserRepository;
+import java.lang.Exception;
+
+public class SessionContext {
+
+    private UserService userService;
+
+    public SessionContext() {
+        userService = UserServiceFactory.getUserService();
+    }
+
+    /**
+    * method that returns the current logged in User or null if
+    * no user is logged in
+    */
+    public User getLoggedInUser() {
+    User currentUser;
+    NonPersistentUserRepository userRepository = new NonPersistentUserRepository();
```

Comment:
Is there a reason we need to create a new NonPersistentUserRepository for every call to getLoggedInUser()?

Consider passing a UserRepository (interface) object in the SessionContext constructor and storing it as a private variable.

Reply:
Done